src/pre_processing/extract_landscape.py

```diff
@@ -0,0 +1,190 @@
+    detector = dlib.get_frontal_face_detector()
+    predictor = dlib.shape_predictor(PREDICTOR_PATH)
+    faces_found = 0
+    faces_not_found = 0
+    logger.debug("Starting facial detection and feature extraction")
+    for i in range(len(photos_matrix)):
+        img = photos_matrix[i, :, :]
+        dets = detector(img, 2)
+        if len(dets) != 0:
+            facial_featutes_matrix[i, :, :] = _get_vectors_of_image_from_image(dets, predictor, img)
+            faces_found += 1
+        else:
+            facial_featutes_matrix[i, :, :] = np.zeros([68, 2])
+            faces_not_found += 1
+        if i % 5000 == 0:
+            logger.debug("Photos gone through {}".format(i))
+
+    logger.debug("Faces found: {}  Faces not found : {}".format(faces_found, faces_not_found))
+    _save_processed_image_matrix(facial_featutes_matrix, only_train_data, only_test_data)
+    return facial_featutes_matrix
+
+
+def _extract_photos_from_file(file_path=DATA_CSV_FILE, extract_first_only=False, only_train_data=False,
+                              only_test_data=False):
+    def _create_image_matrix(only_train_data, only_test_data):
+        if only_train_data:
+            images_matrix = np.zeros([TRAIN_END_POINT, 48, 48], dtype="uint8")
+        elif only_test_data:
+            images_matrix = np.zeros([PUBLIC_TEST_END_POINT - TRAIN_END_POINT, 48, 48], dtype="uint8")
+        else:
+            images_matrix = np.zeros([35887, 48, 48], dtype="uint8")
+        return images_matrix
+
+    logger.debug("Extracting photos from csv")
+    images_matrix = _create_image_matrix(only_train_data, only_test_data)
+
+    with open(file_path, newline='', encoding='utf-8') as f:
+        reader = csv.reader(f)
+        _ = next(reader)
+        for k, pixels in enumerate(reader):
+            k_mod = k
+            image_csv_format = np.zeros([1, IMAGE_SIZE])
+            pixels_formated = [int(a) for a in pixels[1].split(" ")]
+            image_csv_format[0, :] = pixels_formated
+            image_csv_format = np.reshape(image_csv_format, [48, 48])
+            if k % 10000 == 0:
+                logger.debug("photos extracted from csv {}".format(k))
+            if only_train_data and k >= TRAIN_END_POINT - 1:
+                break
+            elif only_test_data:
+                if k <= PUBLIC_TEST_START_POINT:
+                    continue
+                k_mod = k - PUBLIC_TEST_START_POINT
+            images_matrix[k_mod, :, :] = image_csv_format
+            if extract_first_only:
+                break
+
+    logger.debug("Finished extracting photos from csv")
+    return images_matrix
+
+
+def _initialise_gui_window(img):
+    win = dlib.image_window()
+    win.clear_overlay()
+    win.set_image(img)
+    return win
+
+
+def _get_vectors_of_image_from_image(dets, predictor, img, win=None):
+    for k, d in enumerate(dets):
+        shape = predictor(img, d)
+    vec = np.empty([68, 2], dtype=int)
```

Comment:
vector space for points on face. see http://openface-api.readthedocs.io/en/latest/_images/dlib-landmark-mean.png